code/modules/projectiles/item_to_box_mapping.dm

```diff
@@ -0,0 +1,105 @@
+//Pairing of an item and box that contains a number of a given item
+/datum/item_box_pairing
+	var/obj/item/item
+	var/obj/item/box
+	var/items_in_box
+
+//You can have the same item in multiple boxes though...
+/datum/item_to_multiple_box_pairing
+	var/obj/item/item
+	var/list/datum/item_box_pairing/item_box_pairings = list()
+
+//A set of lists that tracks items and their equivalent boxes and vice versa
+/datum/item_to_box_mapping
+	var/list/item_to_box_list = list()
+	var/list/box_to_item_list = list()
+
+/datum/item_to_box_mapping/proc/add_pairing(var/datum/item_box_pairing/ibp)
+	box_to_item_list[ibp.box] = ibp
+	if(!item_to_box_list[ibp.item])
+		item_to_box_list[ibp.item] = new /datum/item_to_multiple_box_pairing()
+		item_to_box_list[ibp.item].item = ibp.item
+	item_to_box_list[ibp.item].item_box_pairings += ibp
+
+
+/datum/item_to_box_mapping/proc/Init()
+	//Ammo magazine boxes, minus loose ammo boxes
+	for(var/obj/item/ammo_box/magazine/AB as anything in typesof(/obj/item/ammo_box/magazine))
```

Comment:
You probably want subtypesof() here

Reply:
No, because `/obj/item/ammo_box/magazine` is the valid ammo box for M41A ammo... Should I fix that?

Reply:
Oh my bad then, nevermind